File: springboot-mall/src/main/java/com/tonyydl/springbootmall/data/dto/UserDTO.java
package com.tonyydl.springbootmall.data.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(toBuilder = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonPropertyOrder({
        "user_id",
        "email",
        "created_date",
        "last_modified_date"
})
public class UserDTO {
    @JsonProperty("user_id")
    private Integer userId;
    private String email;
    @JsonProperty("created_date")
    private long createdDate;
    @JsonProperty("last_modified_date")
    private long lastModifiedDate;
}

File: android-app/app/src/main/java/com/tonyydl/shoppingmallapp/data/dto/UserDTO.kt
package com.tonyydl.shoppingmallapp.data.dto

import com.google.gson.annotations.SerializedName
import com.tonyydl.shoppingmallapp.data.Mapper
import com.tonyydl.shoppingmallapp.data.vo.User

data class UserDTO(
    @SerializedName("user_id") val userId: Int = 0,
    val email: String? = null,
    @SerializedName("created_date") val createdDate: Long? = null,
    @SerializedName("last_modified_date") val lastModifiedDate: Long? = null
) : Mapper<User> {
    override fun transform(): User {
        return User(
            userId = userId,
            email = email.orEmpty(),
            createdDate = createdDate ?: 0L,
            lastModifiedDate = lastModifiedDate ?: 0L
        )
    }
}